internal/actions/test.go

```diff
@@ -0,0 +1,92 @@
+package actions
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	config "github.com/speakeasy-api/sdk-gen-config"
+	"github.com/speakeasy-api/sdk-generation-action/internal/cli"
+	"github.com/speakeasy-api/sdk-generation-action/internal/configuration"
+	"github.com/speakeasy-api/sdk-generation-action/internal/environment"
+	"golang.org/x/exp/slices"
+)
+
+func Test() error {
+	g, err := initAction()
+	if err != nil {
+		return err
+	}
+
+	if _, err = cli.Download("latest", g); err != nil {
+		return err
+	}
+
+	wf, err := configuration.GetWorkflowAndValidateLanguages(false)
+	if err != nil {
+		return err
+	}
+
+	// This will only come in via workflow dispatch, we do accept 'all' as a validate special case
+	var testedTargets []string
+	if providedTargetName := environment.SpecifiedTarget(); providedTargetName != "" {
+		testedTargets = append(testedTargets, providedTargetName)
+	}
+
+	if len(testedTargets) == 0 {
+		files, err := g.GetChangedFilesForPR()
```

Comment:
Maybe worth explaining in a comment what we are doing here and why it's necessary

Reply:
will do